Task_4/scripts/app.js

```diff
@@ -0,0 +1,377 @@
+        }
+      }
+
+      for (key in task) {
+        if (!validateObj[key](task[key])) {
+          throw new Error(ERRORS.valuesNotValidate);
+        }
+      }
+      return true;
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  }
+
+  function addTask(name, description, assignee, status, priority, isPrivate) {
+    try {
+      if (!validateObj.name(name)) {
+        throw new Error(ERRORS.nameNotValidate);
+      }
+      if (!validateObj.description(description)) {
+        throw new Error(ERRORS.descriptionNotValidate);
+      }
+      if (!assignee) {
+        throw new Error(ERRORS.assigneeEmpty);
+      }
+      if (!validateObj.status(status)) {
+        throw new Error(ERRORS.statusNotValidate);
+      }
+      if (!validateObj.priority(priority)) {
+        throw new Error(ERRORS.priorityNotValidate);
+      }
+      if (!validateObj.isPrivate(isPrivate)) {
+        throw new Error(ERRORS.isPrivateNotValidate);
+      }
+
+      const task = {
+        id: generateId(),
+        name,
+        description,
+        createdAt: new Date(),
+        assignee,
+        status,
+        priority,
+        isPrivate,
+        comments: [],
+      };
+
+      tasks.push(task);
+
+      return true;
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  }
+
+  function editTask(
+    id,
+    name = null,
+    description = null,
+    assignee = null,
+    status = null,
+    priority = null,
+    isPrivate = false
+  ) {
+    try {
+      if (arguments.length <= 1) {
+        throw new Error(ERRORS.countAgrumentsNotValidate);
+      }
+
+      if (!isValidTypeId(id)) {
+        throw new Error(ERRORS.invalidValue);
+      }
+
+      if (!findTaskById(id)) {
+        throw new Error(ERRORS.taskNotFound);
+      }
+
+      if (!isValidUser(id)) {
+        throw new Error(ERRORS.userValidation);
+      }
+
+      const task = getTask(id);
+
+      if (name) {
+        if (!validateObj.name(name)) {
+          throw new Error(ERRORS.nameNotValidate);
+        }
+        task.name = name;
+      }
+
+      if (description) {
+        if (!validateObj.description(description)) {
+          throw new Error(ERRORS.descriptionNotValidate);
+        } else {
+          task.description = description;
+        }
+      }
+
+      if (assignee) {
+        if (!validateObj.description(assignee)) {
+          throw new Error(ERRORS.assigneeEmpty);
+        } else {
+          task.assignee = assignee;
+        }
+      }
+
+      if (status) {
+        if (!validateObj.description(status)) {
+          throw new Error(ERRORS.statusNotValidate);
+        } else {
+          task.status = status;
+        }
+      }
+
+      if (priority) {
+        if (!validateObj.description(priority)) {
+          throw new Error(ERRORS.priorityNotValidate);
+        } else {
+          task.priority = priority;
+        }
+      }
+
+      if (isPrivate) {
```

Comment:
if you want change not privated to privated   you won't go into this if{}. You must change conditions into if